cabot_ui/scripts/cabot_keyboard.py

```diff
@@ -91,26 +93,45 @@ def process():
                 lastUp[i] = now
 
         if lastUp[i] is not None and now - lastUp[i] > Duration(seconds=interval):
-            if not hold_active[1]:
+            if not hold_active[1] and upCount[i] > 0:
                 event = cabot_common.event.ClickEvent(buttons=i, count=upCount[i])
-            reset_click_state(i)
+                reset_click_state(i)
 
         # node.get_logger().info(upCount)
         # node.get_logger().info(temp)
         # node.get_logger().info(btnDwn)
 
     if button != -1 and hold_duration > 0:
-        hold_active[button] = True
-        event = cabot_common.event.HoldDownEvent(holddown=button, duration=hold_duration)
-        hold_duration = 0
+        if hold_start_time is None:
+            hold_start_time = now
+            sent_duration = 0
+
+        elapsed_time = (now - hold_start_time).nanoseconds / 1e9
+
+        if elapsed_time > sent_duration + 1.0:
+            sent_duration += 1
+            hold_active[button] = True
+            event = cabot_common.event.HoldDownEvent(holddown=button, duration=int(sent_duration))
+            # node.get_logger().info(f"HoldDownEvent: button={}")
+            msg = std_msgs.msg.String()
+            msg.data = str(event)
+            eventPub.publish(msg)
```

Comment:
You don't need to publish the message here because it will be published in line 130.
The current code will publish a hold-down event twice a second.